nistats/utils.py

```diff
@@ -204,3 +206,159 @@ def pos_recipr(X):
+    version of the file name as a dictionary for each file found to do more
+    detailed operations on the file list.
+
+    Notice that to search in the derivatives folder, it has to be given as
+    part of the main_path. This is useful since the current convention gives
+    exactly the same inner structure to derivatives than to the main BIDS
+    dataset folder, so we can search it in the same way.
+
+    Parameters
+    ----------
+    main_path: str
+        Directory of the BIDS dataset
+
+    file_tag: str accepted by glob, optional (default: '*')
+        The final tag of the desired files. For example 'bold' if one is
+        interested in the files related to the neuroimages.
+
+    file_type: str accepted by glob, optional (default: '*')
+        The type of the desired files. For example to be able to request only
+        'nii' or 'json' files for the 'bold' tag.
+
+    sub_label: str accepted by glob, optional (default: '*')
+        Such a common filter is given as a direct option since it applies also
+        at the level of directories. the label is what follows the 'sub' field
+        in the BIDS convention as 'sub-label'.
+
+    file_folder: str accepted by glob, optional (default: '*')
+        Inside the subject and optional session folders a final level of
+        folders is expected in the BIDS convention that groups files according
+        to different neuroimaging modalities and any other additions of the
+        dataset provider. For example the 'func' and 'anat' standard folders.
+        If given as the empty string '', files will be searched inside the
+        sub-label/ses-label directories.
+
+    filters: list of tuples (str, str), optional (default: [])
+        Filters are of the form (field, label). Only one filter per field
+        allowed. A file for which a filter do not apply will be discarded.
+        Filter examples would be (ses, 01), (variant, smooth) and
+        (task, localizer).
+
+    parse: boolean, optional (default: False)
+        Determines if the function should return a dictionary for each file
+        found with parsed information from the files names to be able to do
+        more complex selection operation on the files.
+
+    sub_folder: boolean, optional (default: True)
+        Determines if the files searched are at the level of
+        subject/session folders or just below the dataset main folder.
+        Setting this option to False with other default values would return
+        all the files below the main directory, ignoring files in subject
+        or derivatives folders.
+
+    allow_other_fields: boolean, optional (default: True)
+        In case filters are provided, this option detemines if the filters
+        provided for the search are supposed to be only fields in the file
+        name. If set to False then any file name that contains additional
+        fields will be removed from the results list.
```

Comment:
Its intended for cases in which one would want to confirm the filters considered accounted for all the fields in the file names. Also when uniqueness of files wants to be enforced (since many files like variants and spaces not specified by the filters might be available), maybe I should handle the uniqueness by throwing an error instead? I used this many times as sanity check when exploring the content of a dataset. It seemed useful at some point, but maybe is too ill defined?